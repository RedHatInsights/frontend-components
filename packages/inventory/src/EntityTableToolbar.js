/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Skeleton, SkeletonSize, PrimaryToolbar } from '@redhat-cloud-services/frontend-components';
import debounce from 'lodash/debounce';

const TEXT_FILTER = 'hostname_or_id';

class EntityTableToolbar extends Component {
    state = {
        textFilter: ''
    }

    debouncedRefresh = debounce((config) => {
        this.props.onRefresh && this.props.onRefresh(config);
    }, 800);

    onSetTextFilter = (value) => {
        const { page, perPage } = this.props;
        this.debouncedRefresh({
            page, perPage, filters: [{
                value: TEXT_FILTER,
                filter: value
            }]
        });
        this.setState({
            textFilter: value
        });
    }

    render() {
        const {
            total,
            page,
            onRefresh,
            perPage,
            filters,
            hasItems,
            pathPrefix,
            apiBase,
            children,
            loaded,
            actionsConfig,
            ...props
        } = this.props;
        return (<PrimaryToolbar
            {...props}
            className="ins-c-inventory__table--toolbar"
            filterConfig={{
                items: [
                    ...!hasItems ? [{
                        label: 'Name',
                        value: 'name-filter',
                        filterValues: {
                            placeholder: 'Find system by name',
                            value: this.state.textFilter,
                            onChange: (_e, value) => this.onSetTextFilter(value)
                        }
                    }] : []
                ]
            }}
            actionsConfig={
                loaded ?
                    actionsConfig :
                    <Skeleton size={SkeletonSize.lg} />
            }
            pagination={loaded ? {
                page,
                itemCount: total,
                perPage,
                onSetPage: (_e, newPage) => onRefresh({ page: newPage, perPage, filters }),
                // eslint-disable-next-line camelcase
                onPerPageSelect: (_e, newPerPage) => onRefresh({ page: 1, per_page: newPerPage, filters })
            } : <Skeleton size={SkeletonSize.lg} />}
        >
            { children }
        </PrimaryToolbar>);
    }
}

EntityTableToolbar.propTypes = {
    total: PropTypes.number,
    filters: PropTypes.array,
    hasItems: PropTypes.bool,
    pathPrefix: PropTypes.number,
    apiBase: PropTypes.string,
    page: PropTypes.number,
    onRefresh: PropTypes.func,
    perPage: PropTypes.number,
    children: PropTypes.node,
    pagination: PrimaryToolbar.propTypes.pagination,
    loaded: PropTypes.bool,
    actionsConfig: PrimaryToolbar.propTypes.actionsConfig
};

function mapStateToProps(
    { entities: { page, perPage, total, loaded, activeFilters } },
    { totalItems, page: currPage, perPage: currPerPage, hasItems }) {
    return {
        page: hasItems ? currPage : page,
        perPage: hasItems ? currPerPage : perPage,
        total: hasItems ? totalItems : total,
        loaded,
        filters: activeFilters
    };
}

export default connect(mapStateToProps)(EntityTableToolbar);
