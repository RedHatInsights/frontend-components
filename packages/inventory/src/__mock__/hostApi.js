import { hosts, tags } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
export const mockTags = new MockAdapter(tags.axios);
