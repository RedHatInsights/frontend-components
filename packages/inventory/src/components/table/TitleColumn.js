/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
import React from 'react';

/**
 * Helper function to proprly calculate what to do when user clicks on first cell.
 * Either full redirect if used with ctrl button or `onRowClick` from props is used.
 * @param {*} event html event, to find out if meta key was clicked.
 * @param {*} key inventory UUID.
 * @param {*} props additional props from `EntityTable` - loaded, onRowClick and noDetail.
 */
const onRowClick = (event, key, { loaded, onRowClick: rowClick, noDetail }) => {
    if (loaded && !noDetail) {
        const isMetaKey = (event.ctrlKey || event.metaKey || event.which === 2);
        if (isMetaKey) {
            const url = new URL(`./${key}`, location.href);
            window.open(url.href);
        } else if (rowClick) {
            rowClick(event, key, isMetaKey);
        }
    }
};

/**
 * Helper function to generate first cell in plain inventory either with clickable detail or just data from attribut.
 * This is later on used in redux in `renderFunc`.
 * @param {React.node} data React node with information that will be shown to user as column title.
 * @param {string} id inventory UUID, used to navigate to correct URL.
 * @param {*} item row data, holds every information from redux store for currecnt row.
 * @param {*} props additional props passed from `EntityTable` - holds any props passed to inventory table.
 */
const TitleColumn = (data, id, item, props) => (
    <div className="ins-composed-col">
        <div>{item?.os_release}</div>
        <div className={props?.noDetail ? 'ins-m-nodetail' : ''}>
            { props?.noDetail ?
                data :
                <a
                    widget="col"
                    href={ `${location.pathname}${location.pathname.substr(-1) === '/' ? '' : '/'}${id}` }
                    onClick={ event => {
                        event.preventDefault();
                        event.stopPropagation();
                        onRowClick(event, id, props);
                    }}
                >
                    { data }
                </a> }
        </div>
    </div>
);

export default TitleColumn;
