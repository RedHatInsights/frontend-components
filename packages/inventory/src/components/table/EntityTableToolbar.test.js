/* eslint-disable camelcase */
import React from 'react';
import EntityTableToolbar from './EntityTableToolbar';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import promiseMiddleware from 'redux-promise-middleware';
import toJson from 'enzyme-to-json';
import { mock, mockTags } from '../../__mock__/hostApi';
import TitleColumn from './TitleColumn';
import debounce from 'lodash/debounce';

jest.mock('lodash/debounce');

describe('EntityTableToolbar', () => {
    let initialState;
    let mockStore;
    beforeEach(() => {
        debounce.mockImplementation(require.requireActual('lodash/debounce'));
        mockStore = configureStore([ promiseMiddleware() ]);
        initialState = {
            entities: {
                activeFilters: [{}],
                loaded: true,
                rows: [{
                    id: 'testing-id',
                    one: 'data'
                }],
                columns: [{ key: 'one', title: 'One', renderFunc: TitleColumn }],
                page: 1,
                perPage: 50,
                total: 500,
                allTags: [
                    { name: 'something', tags: [
                        { count: 5, tag: {
                            namespace: 'something',
                            key: 'some key',
                            value: 'some value'
                        } },
                        { count: 2, tag: {
                            namespace: 'something',
                            key: 'some key',
                            value: null
                        } }
                    ] },
                    {
                        name: 'null',
                        tags: [
                            { count: 2, tag: {
                                namespace: null,
                                key: 'some key',
                                value: null
                            } }
                        ]
                    }
                ]
            }
        };
    });

    describe('DOM', () => {
        it('should render correctly - no data', () => {
            const store = mockStore({
                entities: {
                    loaded: false
                }
            });
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with tags', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar showTags />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
            expect(toJson(wrapper.find('TagsModal'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with items', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar hasItems />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with custom filters', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar filterConfig={{
                    items: [{ label: 'Filter by text' }]
                }} />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with custom activeFilters', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar activeFiltersConfig={{
                    filters: [{
                        category: 'Some',
                        chips: [{
                            name: 'something'
                        }, {
                            name: 'something 2'
                        }]
                    }]
                }} showTags />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with default filters', () => {
            const store = mockStore({
                entities: {
                    ...initialState.entities,
                    activeFilters: [{ staleFilter: [ 'fresh' ] }]
                }
            });
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with default tag filter', () => {
            const store = mockStore({
                entities: {
                    ...initialState.entities,
                    activeFilters: [{ tagFilters: [{
                        type: 'tags', key: 'something', category: 'something', values: [{
                            key: 'some key',
                            group: {
                                label: 'Some tag'
                            },
                            value: 'some value'
                        }]
                    }] }]
                }
            });
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly - with children', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar><div>something</div></EntityTableToolbar>
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render correctly', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar page={1} total={500} perPage={50} />
            </Provider>);
            expect(toJson(wrapper.find('PrimaryToolbar'), { mode: 'shallow' })).toMatchSnapshot();
        });
    });

    describe('API', () => {
        describe('pagination', () => {
            it('should set page ', () => {
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                const store = mockStore(initialState);
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} />
                </Provider>);
                wrapper.find('button[data-action="next"]').first().simulate('click');
                const actions = store.getActions();
                expect(actions.length).toBe(1);
                expect(actions[0]).toMatchObject({ meta: { showTags: false }, type: 'LOAD_ENTITIES_PENDING' });
            });

            it('should set per page ', () => {
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                const store = mockStore(initialState);
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} />
                </Provider>);
                wrapper.find('.pf-c-options-menu__toggle button.pf-c-options-menu__toggle-button').first().simulate('click');
                wrapper.update();
                wrapper.find('ul.pf-c-options-menu__menu button[data-action="per-page-10"]').simulate('click');
                const actions = store.getActions();
                expect(actions.length).toBe(1);
                expect(actions[0]).toMatchObject({ meta: { showTags: false }, type: 'LOAD_ENTITIES_PENDING' });
            });
        });

        describe('delete filter', () => {
            it('should dispatch action on delete filter', () => {
                debounce.mockImplementation(fn => fn);
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
                const store = mockStore(initialState);
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} />
                </Provider>);
                const orignalActionsLength = store.getActions().length;
                wrapper.find('.pf-c-chip-group__list li div button').first().simulate('click');
                wrapper.find('.pf-c-chip-group__list li div button').last().simulate('click');
                const actions = store.getActions();
                expect(actions.length).toBe(orignalActionsLength + 1);
                expect(actions[actions.length - 1]).toMatchObject({ meta: { showTags: false }, type: 'LOAD_ENTITIES_PENDING' });
            });

            it('should remove textual filter', () => {
                debounce.mockImplementation(fn => fn);
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
                const store = mockStore({
                    entities: {
                        ...initialState.entities,
                        activeFilters: [{ value: 'hostname_or_id', filter: 'test' }]
                    }
                });
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} />
                </Provider>);
                wrapper.find('.pf-c-chip-group__list li div button').first().simulate('click');
                const actions = store.getActions();
                expect(actions[actions.length - 1]).toMatchObject({ meta: { showTags: false }, type: 'LOAD_ENTITIES_PENDING' });
            });

            it('should remove tag filter', () => {
                debounce.mockImplementation(fn => fn);
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
                const store = mockStore({
                    entities: {
                        ...initialState.entities,
                        activeFilters: [{ tagFilters: [{
                            type: 'tags', key: 'something', category: 'something', values: [{
                                key: 'some key',
                                group: {
                                    label: 'Some tag'
                                },
                                value: 'some value'
                            }]
                        }] }]
                    }
                });
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} showTags />
                </Provider>);
                wrapper.find('.pf-c-chip-group__list li div button').first().simulate('click');
                const actions = store.getActions();
                expect(actions[actions.length - 1]).toMatchObject({ type: 'ALL_TAGS_PENDING' });
            });

            it('should dispatch action on delete all filters', () => {
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
                const store = mockStore(initialState);
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} />
                </Provider>);
                wrapper.find('.ins-c-chip-filters button.pf-m-link').last().simulate('click');
                const actions = store.getActions();
                expect(actions.length).toBe(2);
                expect(actions[actions.length - 1]).toMatchObject({ type: 'CLEAR_FILTERS' });
            });

            it('should call function on delete filter', () => {
                mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
                mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
                const onDelete = jest.fn();
                const store = mockStore(initialState);
                const wrapper = mount(<Provider store={store}>
                    <EntityTableToolbar page={1} total={500} perPage={50} activeFiltersConfig={{
                        onDelete
                    }} />
                </Provider>);
                wrapper.find('.pf-c-chip-group__list li div button').first().simulate('click');
                expect(onDelete).toHaveBeenCalled();
            });
        });

        it('should call onRefresh', () => {
            const onRefresh = jest.fn((_par, callback) => callback());
            debounce.mockImplementation(fn => fn);
            mock.onGet(/\/api\/inventory\/v1\/hosts.*/).reply(200, {});
            mockTags.onGet(/\/api\/inventory\/v1.*/).reply(200, {});
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={store}>
                <EntityTableToolbar page={1} total={500} perPage={50} onRefresh={onRefresh} />
            </Provider>);
            wrapper.find('.pf-c-options-menu__toggle button.pf-c-options-menu__toggle-button').first().simulate('click');
            wrapper.update();
            wrapper.find('ul.pf-c-options-menu__menu button[data-action="per-page-10"]').simulate('click');
            const actions = store.getActions();
            expect(actions[0]).toMatchObject({ type: 'ENTITIES_LOADING' });
        });
    });
});
