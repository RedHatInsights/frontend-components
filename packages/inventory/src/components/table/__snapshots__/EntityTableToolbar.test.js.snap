// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityTableToolbar DOM should render correctly - no data 1`] = `
<PrimaryToolbar
  actionsConfig={null}
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    <Skeleton
      isDark={false}
      size="lg"
    />
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with children 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with custom activeFilters 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
        Object {
          "category": "Some",
          "chips": Array [
            Object {
              "name": "something",
            },
            Object {
              "name": "something 2",
            },
          ],
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
        Object {
          "filterValues": Object {
            "className": "ins-c-inventory__tags-filter",
            "filterBy": "",
            "items": Array [
              Object {
                "className": "ins-c-inventory__tags-tail",
                "isDisabled": true,
                "label": <span>
                   
                  <Spinner
                    size="md"
                  />
                   
                </span>,
              },
            ],
            "onChange": [Function],
            "onFilter": [Function],
            "selected": Object {},
          },
          "label": "Tags",
          "placeholder": "Filter system by tag",
          "type": "group",
          "value": "tags",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with custom filters 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
        Object {
          "label": "Filter by text",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with default filters 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with default tag filter 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with items 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar ins-c-inventory__table--toolbar-has-items"
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with tags 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
        Object {
          "filterValues": Object {
            "className": "ins-c-inventory__tags-filter",
            "filterBy": "",
            "items": Array [
              Object {
                "className": "ins-c-inventory__tags-tail",
                "isDisabled": true,
                "label": <span>
                   
                  <Spinner
                    size="md"
                  />
                   
                </span>,
              },
            ],
            "onChange": [Function],
            "onFilter": [Function],
            "selected": Object {},
          },
          "label": "Tags",
          "placeholder": "Filter system by tag",
          "type": "group",
          "value": "tags",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": undefined,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": undefined,
      "perPage": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`EntityTableToolbar DOM should render correctly - with tags 2`] = `
<TagsModal
  fetchAllTags={[Function]}
  fetchTagsForSystem={[Function]}
  filters={
    Array [
      Object {},
    ]
  }
  loaded={false}
  onApply={[Function]}
  onToggleTagModal={[Function]}
  pagination={
    Object {
      "page": 1,
      "perPage": 10,
    }
  }
  showTagDialog={false}
  tags={
    Array [
      Object {
        "key": "some key",
        "namespace": "something",
        "value": "some value",
      },
      Object {
        "key": "some key",
        "namespace": "something",
        "value": null,
      },
      Object {
        "key": "some key",
        "namespace": null,
        "value": null,
      },
    ]
  }
  tagsCount={0}
/>
`;

exports[`EntityTableToolbar DOM should render correctly 1`] = `
<PrimaryToolbar
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "category": "Status",
          "chips": Array [
            Object {
              "name": "Fresh",
              "value": "fresh",
            },
            Object {
              "name": "Stale",
              "value": "stale",
            },
          ],
          "type": "staleness",
        },
        Object {
          "category": "Source",
          "chips": Array [
            Object {
              "name": "Insights",
              "value": "insights",
            },
          ],
          "type": "registered_with",
        },
      ],
      "onDelete": [Function],
    }
  }
  className="ins-c-inventory__table--toolbar "
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "onChange": [Function],
            "placeholder": "Filter by name",
            "value": "",
          },
          "label": "Name",
          "value": "name-filter",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Fresh",
                "value": "fresh",
              },
              Object {
                "label": "Stale",
                "value": "stale",
              },
              Object {
                "label": "Stale warning",
                "value": "stale_warning",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "fresh",
              "stale",
            ],
          },
          "label": "Status",
          "type": "checkbox",
          "value": "stale-status",
        },
        Object {
          "filterValues": Object {
            "items": Array [
              Object {
                "label": "Insights",
                "value": "insights",
              },
            ],
            "onChange": [Function],
            "value": Array [
              "insights",
            ],
          },
          "label": "Source",
          "type": "checkbox",
          "value": "source-registered-with",
        },
      ],
    }
  }
  filters={Array []}
  pagination={
    Object {
      "itemCount": 500,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 50,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;
